### PR TITLE
Smells should link to docs for the relevant analyser

### PR DESCRIPTION
**Pull request by ragesoss**
Resolves whitesmith/rubycritic#137

### CODE REVIEW

File: lib/rubycritic/core/smell.rb [hunk 1]
```diff
@@ -47,7 +48,16 @@ def to_json(*a)
     end
 
     def doc_url
-      "https://github.com/troessner/reek/blob/master/docs/#{dasherized_type}.md"
+      case analyser
+      when 'reek'
+        "https://github.com/troessner/reek/blob/master/docs/#{dasherized_type}.md"
+      when 'flay'
+        'http://docs.seattlerb.org/flay/'
```
Comment (troessner) [inline lib/rubycritic/core/smell.rb]: Let's put this string and the next into constants at the top of the file (including ".freeze")

File: lib/rubycritic/core/smell.rb [hunk 2]
```diff
@@ -47,7 +48,16 @@ def to_json(*a)
...
+      when 'flog'
+        'http://docs.seattlerb.org/flog/'
+      else
+        ''
```
Comment (troessner) [inline lib/rubycritic/core/smell.rb]: Let's rather raise a proper runtime error here.

**Comment by troessner** [comment]
2 small comments from my side + fixing specs (that's why travis is red) + squashing to one commit at the end and this bad boy is ready to get merged!

**Comment by Onumis** [comment]
Great! Thanks

**Comment by troessner** [comment]
Great job, thanks @ragesoss

**Pull request merged**